diff -Naru btrfs-progs-v4.0.1-old/cmds-filesystem.c btrfs-progs-v4.0.1-new/cmds-filesystem.c
--- btrfs-progs-v4.0.1-old/cmds-filesystem.c	2015-05-20 22:12:44.000000000 +0900
+++ btrfs-progs-v4.0.1-new/cmds-filesystem.c	2016-03-05 17:20:54.745850661 +0900
@@ -1003,6 +1003,8 @@
 		return BTRFS_COMPRESS_ZLIB;
 	else if (strcmp(optarg, "lzo") == 0)
 		return BTRFS_COMPRESS_LZO;
+	else if (strcmp(optarg, "lz4") == 0)
+		return BTRFS_COMPRESS_LZ4;
 	else {
 		fprintf(stderr, "Unknown compress type %s\n", s);
 		exit(1);
@@ -1013,13 +1015,13 @@
 	"btrfs filesystem defragment [options] <file>|<dir> [<file>|<dir>...]",
 	"Defragment a file or a directory",
 	"",
-	"-v             be verbose",
-	"-r             defragment files recursively",
-	"-c[zlib,lzo]   compress the file while defragmenting",
-	"-f             flush data to disk immediately after defragmenting",
-	"-s start       defragment only from byte onward",
-	"-l len         defragment only up to len bytes",
-	"-t size        minimal size of file to be considered for defragmenting",
+	"-v                 be verbose",
+	"-r                 defragment files recursively",
+	"-c[zlib,lzo,lz4]   compress the file while defragmenting",
+	"-f                 flush data to disk immediately after defragmenting",
+	"-s start           defragment only from byte onward",
+	"-l len             defragment only up to len bytes",
+	"-t size            minimal size of file to be considered for defragmenting",
 	NULL
 };
 
diff -Naru btrfs-progs-v4.0.1-old/ctree.h btrfs-progs-v4.0.1-new/ctree.h
--- btrfs-progs-v4.0.1-old/ctree.h	2015-05-20 22:12:44.000000000 +0900
+++ btrfs-progs-v4.0.1-new/ctree.h	2016-03-05 17:44:44.715879066 +0900
@@ -465,16 +465,17 @@
  * number
  */
 #define BTRFS_FEATURE_INCOMPAT_COMPRESS_LZOv2   (1ULL << 4)
+#define BTRFS_FEATURE_INCOMPAT_COMPRESS_LZ4	(1ULL << 5)
 
 /*
  * older kernels tried to do bigger metadata blocks, but the
  * code was pretty buggy.  Lets not let them try anymore.
  */
-#define BTRFS_FEATURE_INCOMPAT_BIG_METADATA     (1ULL << 5)
-#define BTRFS_FEATURE_INCOMPAT_EXTENDED_IREF	(1ULL << 6)
-#define BTRFS_FEATURE_INCOMPAT_RAID56		(1ULL << 7)
-#define BTRFS_FEATURE_INCOMPAT_SKINNY_METADATA	(1ULL << 8)
-#define BTRFS_FEATURE_INCOMPAT_NO_HOLES		(1ULL << 9)
+#define BTRFS_FEATURE_INCOMPAT_BIG_METADATA     (1ULL << 6)
+#define BTRFS_FEATURE_INCOMPAT_EXTENDED_IREF	(1ULL << 7)
+#define BTRFS_FEATURE_INCOMPAT_RAID56		(1ULL << 8)
+#define BTRFS_FEATURE_INCOMPAT_SKINNY_METADATA	(1ULL << 9)
+#define BTRFS_FEATURE_INCOMPAT_NO_HOLES		(1ULL << 10)
 
 
 #define BTRFS_FEATURE_COMPAT_SUPP		0ULL
@@ -483,6 +484,7 @@
 	(BTRFS_FEATURE_INCOMPAT_MIXED_BACKREF |		\
 	 BTRFS_FEATURE_INCOMPAT_DEFAULT_SUBVOL |	\
 	 BTRFS_FEATURE_INCOMPAT_COMPRESS_LZO |		\
+	 BTRFS_FEATURE_INCOMPAT_COMPRESS_LZ4 |		\
 	 BTRFS_FEATURE_INCOMPAT_BIG_METADATA |		\
 	 BTRFS_FEATURE_INCOMPAT_EXTENDED_IREF |		\
 	 BTRFS_FEATURE_INCOMPAT_RAID56 |		\
@@ -641,8 +643,9 @@
 	BTRFS_COMPRESS_NONE  = 0,
 	BTRFS_COMPRESS_ZLIB  = 1,
 	BTRFS_COMPRESS_LZO   = 2,
-	BTRFS_COMPRESS_TYPES = 2,
-	BTRFS_COMPRESS_LAST  = 3,
+	BTRFS_COMPRESS_LZ4   = 3,
+	BTRFS_COMPRESS_TYPES = 3,
+	BTRFS_COMPRESS_LAST  = 4,
 } btrfs_compression_type;
 
 /* we don't understand any encryption methods right now */
